src/lib/transaction_snark/test/account_timing/account_timing.ml

```diff
@@ -1093,6 +1093,131 @@ let%test_module "account timing check" =
                   then failwithf "Unexpected transaction error: %s" err_str () ) )
 
+    let gen_timed_account_and_transfer ~balance ~min_balance ~amount =
+      let open Quickcheck.Generator.Let_syntax in
+      let timed =
+        let keypair = List.nth_exn keypairs 0 in
+        let balance = Currency.Balance.of_int balance in
+        let nonce = Mina_numbers.Account_nonce.zero in
+        let (timing : Account_timing.t) =
+          Timed
+            { initial_minimum_balance = Currency.Balance.of_int min_balance
+            ; cliff_time = Mina_numbers.Global_slot.of_int 10000
+            ; cliff_amount = Currency.Amount.of_int 100_000_000_000_000
+            ; vesting_period = Mina_numbers.Global_slot.of_int 1000
+            ; vesting_increment = Currency.Amount.of_int 10
+            }
+        in
+        let balance_as_amount = Currency.Balance.to_amount balance in
+        (keypair, balance_as_amount, nonce, timing)
+      in
+      let untimed =
+        let keypair = List.nth_exn keypairs 1 in
+        let balance = Currency.Balance.of_int 200_000_000_000_000 in
+        let nonce = Mina_numbers.Account_nonce.zero in
+        let balance_as_amount = Currency.Balance.to_amount balance in
+        (keypair, balance_as_amount, nonce, Account_timing.Untimed)
+      in
+      let ledger_init_state = Array.of_list [ untimed; timed ] in
+      let parties_command =
+        let open Mina_base in
+        let fee = Currency.Fee.of_int 1_000_000 in
+        let amount = Currency.Amount.of_int amount in
+        let nonce = Account.Nonce.zero in
+        let memo =
+          Signed_command_memo.create_from_string_exn
+            "zkApp transfer, timed account"
+        in
+        let sender_keypair = List.nth_exn keypairs 0 in
+        let zkapp_keypair = List.nth_exn keypairs 1 in
+        let receiver_key =
+          zkapp_keypair.public_key |> Signature_lib.Public_key.compress
+        in
+        let (parties_spec : Transaction_snark.For_tests.Spec.t) =
+          { sender = (sender_keypair, nonce)
+          ; fee
+          ; fee_payer = None
+          ; receivers = [ (receiver_key, amount) ]
+          ; amount
+          ; zkapp_account_keypairs = []
+          ; memo
+          ; new_zkapp_account = false
+          ; snapp_update = Party.Update.dummy
+          ; current_auth = Permissions.Auth_required.Signature
+          ; call_data = Snark_params.Tick.Field.zero
+          ; events = []
+          ; sequence_events = []
+          ; preconditions = None
+          }
+        in
+        Transaction_snark.For_tests.multiple_transfers parties_spec
+      in
+      return (ledger_init_state, parties_command)
+
+    let%test_unit "zkApp command, non-negative balance change, not enough \
+                   balance" =
+      Quickcheck.test
+        ~seed:
+          (`Deterministic
+            "zkapp command, non-negative balance change, min_balance > balance"
```

Comment:
Didn't we agree that we're going to maintain this invariant, so that this case will never happen in practice?

Reply:
Deepthi asked me to write tests for all 3 cases: balance >/=/< min-balance. If that's not possible in practice, let's just discard this one.

Reply:
I meant those cases for new accounts:  balance < min-balance is a failure, balance >= min-balance is valid. (suggested testing the `=` case separately to avoid any off-by-one errors). Since we ensure accounts are created with sufficient min balance, for existing accounts we only need to check that negative balance change doesn't cause balance < min-balance.